client/src/components/RegistrationForm.tsx

```diff
@@ -0,0 +1,77 @@
+import { type FormEvent, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { User } from './UserContext';
+
+/**
+ * Form that registers a user.
+ */
+export function RegistrationForm() {
+  const [isLoading, setIsLoading] = useState(false);
+  const navigate = useNavigate();
+
+  async function handleSubmit(event: FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+    try {
+      setIsLoading(true);
+      const formData = new FormData(event.currentTarget);
+      const userData = Object.fromEntries(formData);
+      const req = {
+        method: 'POST',
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify(userData),
+      };
+      const res = await fetch('/api/auth/sign-up', req);
+      if (!res.ok) {
+        throw new Error(`fetch Error ${res.status}`);
+      }
+      const user = (await res.json()) as User;
+      console.log('Registered', user);
+      console.log(
+        `You can check the database with: psql -d userManagement -c 'select * from users'`
+      );
```

Comment:
Remove all console.logs. You should avoid including any console logs in your pull requests. When you create a pull request, you are saying this is the best I can do, and if your boss sees these, they are going to think you didn't double check your work.

Also you should not be telling the user how to interact with your database, that's a huge security risk